docs/source/topics/scheduler_template.py

```diff
@@ -0,0 +1,182 @@
+    """Base class template for a scheduler."""
+
+    # Query only by list of jobs and not by user
+    _features = {
+        'can_query_by_user': False,
+    }
+
+    # The class to be used for the job resource.
+    _job_resource_class = TemplateJobResource
+
+    _map_status = _MAP_SCHEDULER_AIIDA_STATUS
+
+    def _get_resource_lines(
+        self, num_machines, num_mpiprocs_per_machine, num_cores_per_machine, max_memory_kb, max_wallclock_seconds
+    ):
+        """
+        Return a list of lines (possibly empty) with the header
+        lines relative to:
+
+        * num_machines
+        * num_mpiprocs_per_machine
+        * num_cores_per_machine
+        * max_memory_kb
+        * max_wallclock_seconds
+
+        This is done in an external function because it may change in
+        different subclasses.
+        """
+        return []
+
+    def _get_joblist_command(self, jobs=None, user=None):
+        """The command to report full information on existing jobs."""
+
+        return []  # for instance ['qstat', '-f']
+
+    def _get_detailed_job_info_command(self, job_id):
+        """Return the command to run to get the detailed information on a job,
+        even after the job has finished.
+
+        The output text is just retrieved, and returned for logging purposes.
+        """
+        return '' # for instance f'tracejob -v {escape_for_bash(job_id)}'
+
+    def _get_submit_script_header(self, job_tmpl):
+        """Return the submit script header, using the parameters from the job_tmpl.
+
+        Args:
+           job_tmpl: an JobTemplate instance with relevant parameters set.
+
+        """
+        return ''
+
+    def _get_submit_command(self, submit_script):
+        """Return the string to submit a given script.
+
+        Args:
+            submit_script: the path of the submit script relative to the working
+                directory.
+                IMPORTANT: submit_script should be already escaped.
+        """
+        submit_command = '' # for instance f'qsub {submit_script}'
+
+        _LOGGER.info(f'submitting with: {submit_command}')
+
+        return submit_command
+
+    def _parse_joblist_output(self, retval, stdout, stderr):
+        """Parse the queue output string, as returned by executing the command returned by
+        `_get_joblist_command command`.
+
+        Return a list of JobInfo objects, one of each job, each relevant parameters implemented.
+
+        Note: depending on the scheduler configuration, finished jobs may
+            either appear here, or not.
+            This function will only return one element for each job find
+            in the qstat output; missing jobs (for whatever reason) simply
+            will not appear here.
+        """
+        return []
+
+    @staticmethod
+    def _convert_time(string):
+        """Convert a time string to a number of seconds."""
+        hours = 0
+        mins = 0
+        secs = 20
+
+        return hours * 3600 + mins * 60 + secs
```

Comment:
What is this function doing here? It is not used, and it also seems completely wrong? Probably just remove it.